jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/UrlEncoded.java

```diff
@@ -329,7 +329,7 @@ private static void decodeUtf8To(String query, int offset, int length, BiConsume
             switch (c)
             {
                 case '&':
-                    value = buffer.takeCompleteString(() -> new IllegalArgumentException("Invalid value: Bad UTF-8"));
+                    value = buffer.takeCompleteString(() -> new Utf8StringBuilder.InvalidUtf8Exception("Invalid value: Bad UTF-8"));
```

Comment:
I would prefer something like this instead ...

``` java
value = buffer.takeCompleteString(() -> new IllegalArgumentException(
new CharacterCodingException("Invalid value: Bad UTF-8")));
```

Reply:
The problem is that `CharacterCodingException` does not take a message.  It doesn't even take a cause, so we really have to extend it and force the toString and message, otherwise we don't know what the character coding problem is.

So now I have:
```java

    public static class Utf8CharacterCodingException extends CharacterCodingException
    {
        @Override
        public String getMessage()
        {
            return "Invalid UTF-8";
        }

        @Override
        public String toString()
        {
            return "%s@%x: Invalid UTF-8".formatted(CharacterCodingException.class.getSimpleName(), hashCode());
        }
    }

    public static class Utf8IllegalArgumentException extends IllegalArgumentException
    {
        public Utf8IllegalArgumentException()
        {
            super(new Utf8CharacterCodingException());
        }
    }
```

So any exception will always have a `CharacterCodingException` in its causal chain. It will also have the "Invalid ITF-8" message reported.